File: 560_sum_of_subarrays.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count=0
        _map= {}
        curr_sum=0
        if(len(nums)==0):
            return 0
       
        
        for n in range(len(nums)):
            curr_sum+=nums[n]
            if(curr_sum==k):
                count+=1
            if((curr_sum-k) in _map):
                count+=_map[curr_sum-k]
            if(curr_sum in _map):
                _map[curr_sum]+=1
            else:
                _map[curr_sum] =1
        return count; 
        

File: 560_sum_of_subarrays.java
class Solution {
    public int subarraySum(int[] nums, int k) {
       if(nums.length==0) return 0;
        
        int count=0;
        
       HashMap<Integer,Integer> _map= new HashMap<Integer,Integer>();
        
        int curr_sum=0;
       for(int n: nums){
           curr_sum+=n;
           
           if(curr_sum==k){
               count++;
           }
           
           if(_map.containsKey(curr_sum-k)){
               count+=_map.get(curr_sum-k);
           }
           
           if(_map.containsKey(curr_sum)){
               int val= _map.get(curr_sum)+1;
               _map.replace(curr_sum,val);
           }else
           {
               _map.put(curr_sum,1);
           }
       }
        return count; 
    }
}
